docs/guide/arrays.md

```diff
@@ -0,0 +1,71 @@
+# Arrays
+
+HyperFormula lets you perform operations on arrays.
+
+You can:
+- Operate on arrays just like on non-array values
+- Pass arrays (instead of values) to functions
+- Constrain the output array's size
+- Filter an array based on boolean arrays
+
+## About arrays in HyperFormula
+
+There are two types of arrays in HyperFormula:
+* **Range**: a range of cell addresses (e.g. `A1:A10`)
+* **Ad-hoc array**: an array with no ground-truth range that defines the values (e.g. `5*A1:B5`)
+
+Non-array values are **scalars**.
+
+## Operating on arrays
+
+When you perform an operation on an array, each output array value is a result of that operation on the corresponding input array value.
+
+To enable array arithmetic for your formula, use the [`ARRAYFORMULA`](built-in-functions.md#array-manipulation) function:
```

Comment:
Can be configured so it does not require ARRAYFORMULA function:
https://github.com/handsontable/hyperformula/blob/1c6a04ee21c65cc3f5a6e366db754d8cc1be616c/src/Config.ts#L332-L339

Reply:
@wojciechczerniak added: 05151bc